File: go/gigasecond/gigasecond.go
// Package gigasecond determines the date and time one gigasecond after a date.
package gigasecond

// import path for the time package from the standard library
import "time"

// AddGigasecond returns the time one thousand million seconds after given time.
func AddGigasecond(t time.Time) time.Time {
	const gigasecond time.Duration = 1_000_000_000 * time.Second
	return t.Add(gigasecond)
}

File: javascript/gigasecond/gigasecond.js
/**
 * Calculates a date that is one gigasecond (1,000,000,000 seconds) after the provided date.
 * 
 * @param {Date} date - The date to calculate the gigasecond from. 
 * @returns {Date} A new Date object representing the date one gigasecond after the input date. 
 */
export const gigasecond = (date) => new Date(date.getTime() + 1e12);